### DISCUSSION TITLE
The latest wagmi 1.0 is icompatible with the react installation instructions

### DISCUSSION THREAD

**Discussion by paintoshi** [WalletConnect/walletconnect-monorepo#2328] [Developer Support] [answered]
As per https://docs.walletconnect.com/2.0/web3modal/react/installation is to install latest wagmi. But the latest 1.0 has changed quite a bit. They migrated from ethers to viem which affects how the createClient, configureChains and other methods are used. Can be found in the migration guide https://next.wagmi.sh/react/migration-guide#1xx-breaking-changes.

Need to either change the guide to wagmi@0.12.12 or update the config instructions. Wagmi 0.12.13 uses @walletconnect/ethereum-provider 2.7.2 which will fail to build in React due to an invalid reference to a react-native package. Wagmi 0.12.12 uses ethereum-provider 2.7.0, which don't have that issue. See https://github.com/WalletConnect/walletconnect-monorepo/issues/2293

**Comment by xzilja** [comment, marked as answer]
We just released `2.4.0` of web3modal and updated the docs for usage with wagmi v1 👍
